### PR TITLE
fix: Add the CompressedTensorsW8A16Fp8 scheme to support the extension of …

### PR DESCRIPTION
**Pull request by liwenju0**
…quantized tensors.

## Motivation

fix this issue: https://github.com/sgl-project/sglang/issues/5135

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/marlin_fp8_utils.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: Apache-2.0
+from typing import Optional
+
+import torch
+
+try:
+    import vllm
+    from vllm import _custom_ops as ops
```
Comment (BBuf) [inline python/sglang/srt/layers/quantization/marlin_fp8_utils.py]:
```suggestion
    from vllm import _custom_ops as vllm_ops
```

File: python/sglang/srt/layers/quantization/marlin_fp8_utils.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    size_n: int,
+    size_k: int,
+    bias: Optional[torch.Tensor],
+) -> torch.Tensor:
+    # For GPUs that lack FP8 hardware support, we can leverage the
+    # Marlin kernel for fast weight-only FP8 quantization
+
+    reshaped_x = input.reshape(-1, input.shape[-1])
+    out_shape = input.shape[:-1] + (size_n, )
+
```
Comment (BBuf) [inline python/sglang/srt/layers/quantization/marlin_fp8_utils.py]:
```suggestion
if not VLLM_AVAILABLE:
                raise ImportError(
                    "vLLM is not installed, to use CompressedTensorsW8A16Fp8, please install vLLM"
                )
```

**Review by BBuf — APPROVED** [review]
LGTM.

File: python/sglang/srt/layers/quantization/marlin_fp8_utils.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+) -> torch.Tensor:
+    # For GPUs that lack FP8 hardware support, we can leverage the
+    # Marlin kernel for fast weight-only FP8 quantization
+
+    reshaped_x = input.reshape(-1, input.shape[-1])
+    out_shape = input.shape[:-1] + (size_n, )
+if not VLLM_AVAILABLE:
+                raise ImportError(
+                    "vLLM is not installed, to use CompressedTensorsW8A16Fp8, please install vLLM"
+                )
```
Comment (olliestanley) [inline python/sglang/srt/layers/quantization/marlin_fp8_utils.py]: Indentation looks broken here

Comment (liwenju0) [reply python/sglang/srt/layers/quantization/marlin_fp8_utils.py]: yes, i have resovled this error.
